src/program/lwaftr/run/run.lua

```diff
@@ -130,13 +130,10 @@ end
 -- Requires a V4V6 splitter if running in on-a-stick mode and VLAN tag values
 -- are the same for the internal and external interfaces.
 local function requires_splitter (opts, conf)
-   local device, id, queue = lwutil.parse_instance(conf)
-   if opts["on-a-stick"] then
-      local internal_interface = queue.internal_interface
-      local external_interface = queue.external_interface
-      return internal_interface.vlan_tag == external_interface.vlan_tag
-   end
-   return false
+   local queue = select(3, lwutil.parse_instance(conf))
```

Comment:
OMG I had no idea that "select" existed.  Weird!

I think you're probably right here, but I don't understand the whole picture; is it possible to configure two MAC addresses on one interface and use VMDq to present them as separate interfaces?  I think so, right?  In that case if the vlan tags are the same, no splitter needed.

Likewise we should allow queue.external_interface.device to be set, right?  It just *defaults* to be the same as the internal interface.  But we should allow users to specify it as being the same as the external interface.  In that case we should also go through the bump-in-the-wire configuration.  Right?

Reply:
lol i said "right?" many times.  that means i have little confidence in what i am saying :)

Reply:
The scenarios are the following when running on-a-stick:

- Both external and internal interfaces have the same VLAN tag value => a V4V6 splitter is required to separate incoming traffic to their corresponding apps.
- External and internal interfaces use different VLAN tags. Since we rely on the NIC to add and strip VLAN tag values, we need to virtualize the NIC so we can use different VLAN tag values per virtual NIC. Since we are already creating two virtual NICs (inet-side and b4-side), we don't the splitter. 

I don't understand the `queue.external_interface.device` question. AFAIK, it's possible to set this value in a configuration. If an user doesn't set this value in a configuration, then it's nil. The value is only set if --v6 is passed as an argument.

In the former code, a splitter was not needed if we were in non on-a-stick mode (that's why the argument was checked), but now we assume this function is only called if we're going to run on-a-stick (we know we're in on-a-stick mode because `queue.external_interface.device` is nil).

I agree it sounds a bit confusing, but I think it makes sense (I'm not 100% sure about `queue.external_interface.device`, maybe there's something more to do there).

Reply:
The value can also be set if it's set in the configuration file

Reply:
OK, I think I understand now. I assumed that if `queue.external_interface.device` was set we are in a `bump-in-the-wire` scenario (non on-a-stick), but it could be the case that an user sets this value to the same value as the internal interface, which will be an `on-a-stick` scenario. I will check what happens in all these cases.